Fix CaltechHPC environment

nilsvu opened the pull request:
- The previous environment had an issue with loading numeric initial data (took ~40min, now takes 16 seconds).
- It also ran BBH evolutions extremely slowly for some reason (2M in 24 hours).
- It was actually built for cascadelake, not skylake.

Kyle and I reinstalled the environment on skylake, so we should be able to run on all available nodes on CaltechHPC now. I confirmed that the IO issues are fixed and a BBH runs at reasonable speed.

## Proposed changes

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu committed:
Fix CaltechHPC environment

- The previous environment had an issue with loading numeric
  initial data (took ~40min, now takes 16 seconds).
- It also ran BBH evolutions extremely slowly for some reason.
- It was actually built for cascadelake, not skylake.

Kyle and I reinstalled the environment on skylake, so we
should be able to run on all available nodes on CaltechHPC
now. I confirmed that the IO issues are fixed and a BBH runs
at reasonable speed.

nilsvu commented:
Marking priority because BBH evolutions are essentially broken on CaltechHPC at the moment.

knelli2 commented:
I'm testing this now. I'll report back once I'm confident everything is working smoothly (or as non-roughly as possible)

knelli2 commented on support/Environments/caltech_hpc_gcc_skylake.sh:
```diff
@@ -28,21 +28,32 @@ spectre_run_cmake() {
+    # - We temporarily turn of ParaView until a compatible version is installed.
     cmake -D CMAKE_C_COMPILER=gcc \
           -D CMAKE_CXX_COMPILER=g++ \
           -D CMAKE_Fortran_COMPILER=gfortran \
           -D CHARM_ROOT=$CHARM_ROOT \
+          -D BLA_VENDOR=OpenBLAS \
           -D CMAKE_BUILD_TYPE=Release \
+          -D BUILD_DOCS=OFF \
+          -D DEBUG_SYMBOLS=OFF \
           -D MEMORY_ALLOCATOR=JEMALLOC \
           -D BUILD_PYTHON_BINDINGS=ON \
           -D MACHINE=CaltechHpcSkylake \
           -D OVERRIDE_ARCH=skylake \
-          -D ENABLE_PARAVIEW=ON \
+          -D ENABLE_PARAVIEW=OFF \
```
Were there issues finding paraview? This is really only an issue for the CLI so not an immediate priority.

nilsvu replied:
No issues, I sent a request to the CaltechHPC admins to build ParaView with the same Python etc as the rest of the build. They haven't responded yet.

knelli2 commented on support/Environments/caltech_hpc_gcc_skylake.sh:
```diff
@@ -13,12 +13,12 @@ spectre_setup_modules() {
 
 spectre_load_modules() {
```
[random line] Should we remove the `caltech_hpc_gcc_icelake.sh` env now and rename this one to just `caltech_hpc_gcc.sh`? I think that'll be less confusing for users.

nilsvu replied:
Yes I think so. Now or later?

knelli2 replied:
I'd say now. Just rip the Band-Aid off. Be sure to also change the cluster installation instructions, maybe adding a little description that whatever node you build on, you can only run on that type of node or newer.

nilsvu replied:
ok done

knelli2 reviewed (commented):
Ok I have tested this out and am now confident that this will work on all nodes of CaltechHPC (skylake, cascadelake, icelake).

Note that we had to add the environment variable `FI_PROVIDER=tcp` to the `skylake-2024-04` environment because the mpi module kept trying to use UCX for the fabric but because of #3886, we can't use UCX, so we go with TCP instead.

nilsvu committed:
Schedule.py: support passing extra params as dict

This allows extra params like `-p reservation=sxs`
passed to BBH pipeline commands.
